### ISSUE TITLE
Minimal example fails to build - archiver error

### ISSUE THREAD

**Issue by ohmree** [Boscop/web-view#244]
I'm unable to build even the simplest project with web-view, as the build script from webview-sys crashes with an error I couldn't find anywhere on the issue tracker - `ar` can't find `libwebview.a`.  
My `main.rs` is just the minimal wikipedia example, and my `Cargo.toml` only has web-view as a dependency.  
I'm on Arch Linux, webkit2gtk is installed and `pkg-config` can find it:

```
$ pkg-config --cflags --libs webkit2gtk-4.0
-I/usr/include/webkitgtk-4.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/gio-unix-2.0 -I/usr/include/cloudproviders -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/at-spi-2.0 -I/usr/include/libsoup-2.4 -pthread -I/usr/include/libxml2 -lwebkit2gtk-4.0 -lgtk-3 -lgdk-3 -lz -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lsoup-2.4 -lgio-2.0 -ljavascriptcoregtk-4.0 -lgobject-2.0 -lglib-2.0 
```

Here's the error I'm getting:

```
$ cargo build
   Compiling proc-macro2 v1.0.24
   # ...
   Compiling soup-sys v0.10.0
The following warnings were emitted during compilation:

warning: ar: '/path/to/my/project/target/debug/build/webview-sys-d9d52115717c508a/out/libwebview.a': No such file

error: failed to run custom build command for `webview-sys v0.6.0`

Caused by:
  process didn't exit successfully: `/path/to/my/project/target/debug/build/webview-sys-5be1725bb6ce4ad6/build-script-build` (exit code: 1)
  --- stdout
  cargo:rerun-if-env-changed=WEBKIT2GTK_4.0_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=WEBKIT2GTK_4.0_STATIC
  cargo:rerun-if-env-changed=WEBKIT2GTK_4.0_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=SYSROOT
  cargo:rerun-if-env-changed=WEBKIT2GTK_4.0_STATIC
  cargo:rerun-if-env-changed=WEBKIT2GTK_4.0_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rustc-link-search=native=/usr/lib
  cargo:rustc-link-lib=webkit2gtk-4.0
  cargo:rustc-link-lib=gtk-3
  cargo:rustc-link-lib=gdk-3
  cargo:rustc-link-lib=z
  cargo:rustc-link-lib=pangocairo-1.0
  cargo:rustc-link-lib=pango-1.0
  cargo:rustc-link-lib=harfbuzz
  cargo:rustc-link-lib=atk-1.0
  cargo:rustc-link-lib=cairo-gobject
  cargo:rustc-link-lib=cairo
  cargo:rustc-link-lib=gdk_pixbuf-2.0
  cargo:rustc-link-lib=soup-2.4
  cargo:rustc-link-lib=gio-2.0
  cargo:rustc-link-lib=javascriptcoregtk-4.0
  cargo:rustc-link-lib=gobject-2.0
  cargo:rustc-link-lib=glib-2.0
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=WEBKIT2GTK_4.0_STATIC
  cargo:rerun-if-env-changed=WEBKIT2GTK_4.0_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  TARGET = Some("x86_64-unknown-linux-gnu")
  HOST = Some("x86_64-unknown-linux-gnu")
  AR_x86_64-unknown-linux-gnu = None
  AR_x86_64_unknown_linux_gnu = None
  HOST_AR = None
  AR = None
  running: "ar" "s" "/path/to/my/project/target/debug/build/webview-sys-d9d52115717c508a/out/libwebview.a"
  cargo:warning=ar: '/path/to/my/project/target/debug/build/webview-sys-d9d52115717c508a/out/libwebview.a': No such file
  exit code: 1

  --- stderr


  error occurred: Command "ar" "s" "/path/to/my/project/target/debug/build/webview-sys-d9d52115717c508a/out/libwebview.a" with args "ar" did not execute successfully (status code exit code: 1).


warning: build failed, waiting for other jobs to finish...
error: build failed
```

It looks to me like a bug but I'm not 100% sure. Either way I'd love to get some help so I can use this library, it's perfect for my use case.

Thanks in advance for any help!

**Comment by arm32x** [comment]
I'm having the same issue running Manjaro (based on Arch Linux). I'm using the nightly toolchain.

**Comment by ohmree** [comment]
I forgot to mention that I'm also using the nightly toolchain, but switching to stable made no difference.

**Comment by pps5** [comment]
I faced same issue on Manjaro and finally I found the workaround:

1. clone this repository
2. add return just after pkg_config check (https://github.com/Boscop/web-view/blob/master/webview-sys/build.rs#L44).

In webview-sys/build.rs, build target file is not specified, just checking whether "webkit2gtk-4.0" exists, if the system name contains "linux" or "bsd". It looks like cc crate try to archives build result, but it cannot find the result due to the build script.

But this crate works before I updated system packages, so probably some package behavior changed by the update.

**Comment by helgoboss** [comment]
Having the same issue now.

@pps5 Not sure if I understand your workaround. If you add a `return`, it will not compile the webview C code, will it?

**Comment by gBopHuk** [comment]
If I understand correctly the last commit means that we don't need the compilation because of 
"With this, Linux gtk backend is now fully in Rust, no c compiler needed."

https://github.com/Boscop/web-view/commit/44ed909ea3defb29d8d786e25d5bb26d379ee8c3#diff-a0f6d16f755a3f21a188abed938d6f39ddc41966d7f59899a7a3f58dea7dda68

**Comment by pps5** [comment]
@helgoboss 
yes, it won't compile and I think what @gBopHuk said is correct. GTK APIs should be provided by gtk module.

**Comment by zxey** [comment]
Published new patch version 0.7.2 with this fix.
